src/gui/log/logmodel.cpp

```diff
@@ -144,32 +154,14 @@ void LogMessageModel::handleNewMessage(const Log::Msg &message)
 {
     const QString time = QDateTime::fromMSecsSinceEpoch(message.timestamp).toString(Qt::SystemLocaleShortDate);
     const QString messageText = message.message;
-
-    QColor foreground;
-    switch (message.type) {
-    // The RGB QColor constructor is used for performance
-    case Log::NORMAL:
-        foreground = QApplication::palette().color(QPalette::WindowText);
-        break;
-    case Log::INFO:
-        foreground = QColor(0, 0, 255); // blue
-        break;
-    case Log::WARNING:
-        foreground = QColor(255, 165, 0);  // orange
-        break;
-    case Log::CRITICAL:
-        foreground = QColor(255, 0, 0);  // red
-        break;
-    default:
-        Q_ASSERT(false);
-        break;
-    }
+    const QColor foreground = m_foregroundForMessageTypes[message.type];
```

Comment:
We're still getting complains about laggy log widget, would it make sense to use switch case here by not using hash?

Reply:
>We're still getting complains about laggy log widget

Where?

Reply:
https://github.com/qbittorrent/qBittorrent/issues/12740

Reply:
>#12740

Looks to me that the reporter is referring to "Blocked IPs" widget not the log widget.

Reply:
> Switch() is almost always faster than hash lookups.

yes that's why I was asking would it make sense to remove `m_foregroundForMessageTypes` and use switch case here

Reply:
>yes that's why I was asking would it make sense to remove m_foregroundForMessageTypes and use switch case here

Well I mean there is no strong opinion from me, it is up to you.

Reply:
I will just leave it as it is for now.